packages/chevrotain/src/api-type-checking.ts

```diff
@@ -0,0 +1,21 @@
+import * as apiDefs from "../api"
```

Comment:
There is no need for this file to be part of the src directory anymore.
Could we move it to one of the tests directories?
Perhaps the definitions integration tests?

See:
- https://github.com/SAP/chevrotain/tree/master/packages/chevrotain/test_integration/definitions
- https://github.com/SAP/chevrotain/blob/master/packages/chevrotain/package.json#L69-L71

Reply:
Sorry for replying late. I agree with putting this file into tests. Do we need to extract a generic type `AssertApiImplementation<T>` in a `assert-api.ts` and then import this type and use it to test the es6-module and the `chevrotain` namespace?

Reply:
I may be misunderstanding something, are you suggesting additional changes needed for this file to work as an extracted test? 

I think we can just **copy** it to one of the tests folders so it would "execute" the desired tests (via tsc compilation) and thus we will avoid adding it to the npm package and have better separation of concerns (tests vs prod code).